src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SearchScope.cs

```diff
@@ -61,28 +64,73 @@ protected SearchScope(bool exact, CancellationToken cancellationToken)
             }
         }
 
-        private class ProjectSearchScope : SearchScope
+        private abstract class ProjectSearchScope : SearchScope
         {
-            private readonly bool _includeDirectReferences;
-            private readonly Project _project;
+            protected readonly Project _project;
 
-            public ProjectSearchScope(Project project, bool includeDirectReferences, bool ignoreCase, CancellationToken cancellationToken)
+            public ProjectSearchScope(Project project, bool ignoreCase, CancellationToken cancellationToken)
                 : base(ignoreCase, cancellationToken)
             {
                 _project = project;
-                _includeDirectReferences = includeDirectReferences;
+            }
+
+            public override SymbolReference CreateReference<T>(SearchResult<T> searchResult)
+            {
+                return new ProjectSymbolReference(
+                    searchResult.WithSymbol<INamespaceOrTypeSymbol>(searchResult.Symbol), _project.Id);
+            }
+        }
+
+        private class ProjectAndDirectReferencesSearchScope : ProjectSearchScope
+        {
+            public ProjectAndDirectReferencesSearchScope(Project project, bool ignoreCase, CancellationToken cancellationToken)
+                : base(project, ignoreCase, cancellationToken)
+            {
             }
 
             protected override Task<IEnumerable<ISymbol>> FindDeclarationsAsync(string name, SymbolFilter filter, SearchQuery searchQuery)
             {
-                return SymbolFinder.FindDeclarationsAsync(
-                    _project, searchQuery, filter, _includeDirectReferences, cancellationToken);
+                return SymbolFinder.FindDeclarationsAsync(_project, searchQuery, filter, cancellationToken);
             }
+        }
 
-            public override SymbolReference CreateReference<T>(SearchResult<T> searchResult)
+        private class ProjectSourceOnlySearchScope : ProjectSearchScope
```

Comment:
Same for how this one is optimized for projects not directly referenced by the "current project"